src/consumer.ts

```diff
@@ -142,7 +143,36 @@ export class Consumer extends TypedEventEmitter {
       this.emit('aborted');
```

Comment:
I wonder what people's thoughts might be on if abort should also wait for polling to complete? Side thought mainly, possibly one for a discussion, just a note :).